background/rust.tex

```diff
@@ -331,80 +415,87 @@ \subsubsection{Lifetimes}
+This makes it possible to describe and utilize many powerful idioms and paradigms, and concurrency models that are based on these core features of the language.
+Thus, Rust's way of providing concurrency is made possible because the ownership system guarantees that it will be safe.
+Indeed, the core concurrency functionality in Rust is merely an abstraction that is defined in the standard library, is is not a feature in the language itself.
+This is an important distinction, because it allows for any number 3rd party concurrency libraries to evolve based on the same core principle of ownership, without being dependent on the concurrency idioms provided in the standard library.
 
 \subsection{Unsafe Code} % (fold)
 \label{ssub:unsafe_code}
 
-The strong type system along with lifetime inference and borrow-checking help Rust become a safe language.
-The language provides safe, zero-cost abstractions over the machine it is running on, but it is not always possible to guarantee full safety without sacrificing performance, or vice versa.
-Achieving good performance is often a synonym with direct memory access and pointer manipulation, which in itself is not considered to be safe operations.
-\todo{At least by Rust. Depends on who considers it.}
+Rust's strong type system and static guarantees about memory safety goes a long way of verifying the safety of programs, but for certain programs the restrictions set by the compiler might be too conservative.
+There exists many programs that are indeed safe, but still not possible for the compiler to verify.
+Examples of code that the compiler cannot verify is code that utilize \emph{raw pointers}, i.e. pointers to mutable data (like a C pointer), code that call directly \emph{into} external C libraries through Rust's \gls{ffi}, or code that can be called \emph{from} \gls{ffi}.
+
+The ability to make new abstractions that optimize away to essentially nothing is an important feature of Rust.
+Sometimes this means that the underlying code needs direct memory access and the ability to dereference raw pointers or call in to external libraries.
+For these tasks, Rust provides a new keyword, \texttt{unsafe}.
+It is important to note that this \texttt{unsafe} keyword is introduced in order to \emph{keep} Rust's semantics about memory safety, when the compiler can no longer guarantee it, it is up to the programmer to tell the compiler that the code inside the \texttt{unsafe} block is indeed safe.
+\autoref{tab:reasons_for_unsafe} summarizes the three things that are possible to do in an \texttt{unsafe} block, and why the operations are considered unsafe.
+It is important to notice that Rust's borrow checker is still active across the unsafe code, and that it will still issue errors during compilation if the rules about ownership is not withheld.
```

Comment:
`are not withheld`. 

Can you explain this a bit further? This is the fact that the sentence `the borrow checker is disabled in unsafe section` is false, right?